### PR TITLE
capture thisArg of optionalChaining in parens

### PR DESCRIPTION
**Pull request by ajafff**
Fixes: #35476

@rbuckton for review

### CODE REVIEW

File: tests/baselines/reference/parentheses.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+(o1)(o1 ?? 1);
+(o2?.b)(o1 ?? 1);
+(o3?.b())(o1 ?? 1);
+(o4?.b().c)(o1 ?? 1);
+
+
+//// [parentheses.js]
+var _a;
+(o1)(o1 !== null && o1 !== void 0 ? o1 : 1);
+(o2 === null || o2 === void 0 ? void 0 : o2.b).call(o2, o1 !== null && o1 !== void 0 ? o1 : 1);
```
Comment (jridgewell) [inline tests/baselines/reference/parentheses.js]: Nit: this doesn’t need a `.call`:

```js
o2 === null || o2 === undefined ? void 0 : o2.b(o1 ?? 1);

Comment (ajafff) [reply tests/baselines/reference/parentheses.js]: But that wouldn't throw an error if `o2` is nullish.

Comment (jridgewell) [reply tests/baselines/reference/parentheses.js]: Ohh, it's wrapped in parens. Never mind, forgot that made the call non-optional.

**Comment by rbuckton** [comment]
@ajafff can you update this PR? Optional chaining has moved from _esnext.ts_ to _es2020.ts_ and the factory function calls need to be updated to use `factory.` to use the new `NodeFactory` API.

**Comment by rbuckton** [comment]
I've gone ahead and made the necessary updates. Once CI passes I will merge.

**Pull request merged**